### PR TITLE
Use command handler to handle deploy command failed

### PR DESCRIPTION
**Pull request by kiya956**
impact:
    sanity/agent/agent.py
    sanity/agent/deploy.py

description:
    reduce duplicate code.

test:
    N/A

### CODE REVIEW

File: sanity/agent/deploy.py [hunk 1]
```diff
@@ -183,13 +183,6 @@ def deploy(con, method, user_init, update_boot_assets, timeout=600):
     match method:
         case "uuu":
             if syscmd("sudo uuu uc.lst") != 0:
-                mail.send_mail(
-                    FAILED,
-                    (
-                        f"{dev_data.project} auto sanity was failed, "
-                        "deploy failed."
-                    ),
-                )
```
Comment (tsunghanliu) [inline sanity/agent/deploy.py:192]: Can `send_mail()` in `init_mode_login()` be removed as well? Then we can remove `from sanity.agent.mail import mail` from `deploy.py`.

**Review by tsunghanliu — COMMENTED** [review]
one question about removing `send_mail()` entirely

File: sanity/agent/deploy.py (hunk 1 above)
Comment (kiya956) [reply sanity/agent/deploy.py:192]: in here we have two different case, one is init to login timeout, another is get serial assertion timeout. so if we remove the send_mail here, we need more return value to know what is the failed case

File: sanity/agent/agent.py [hunk 2]
```diff
@@ -38,21 +66,22 @@ def start(plan, con, sched=None):
...
                 elif "deploy" in stage.keys():
+                    status = deploy(
+                        con,
+                        stage["deploy"].get("utility"),
+                        stage["deploy"].get("method"),
+                        stage["deploy"].get("update_boot_assets", False),
+                        stage["deploy"].get("timeout", 600),
+                    )
+
                     print(gen_head_string("deploy procedure"))
```
Comment (tsunghanliu) [inline sanity/agent/agent.py:77]: guess this prompt should be printed before `status = deploy(...)`

File: sanity/agent/deploy.py [hunk 3]
```diff
@@ -469,9 +487,9 @@ def init_mode_login(con, user_init, timeout=600):
...
-        mail.send_mail(
-            FAILED,
-            f"{dev_data.project} auto sanity was failed."
-            "Initial Device timeout: connect to store timeout",
-        )
-        return FAILED
+        return {
+            "code": FAILED,
+            "mesg": f"{dev_data.project} auto sanity was failed."
+            " Initial Device timeout: connect to store timeout",
```
Comment (tsunghanliu) [inline sanity/agent/deploy.py:493]: Is the space here in purpose?

**Review by tsunghanliu — CHANGES_REQUESTED** [review]
LGTM, but two nitpicks.

File: sanity/agent/deploy.py (hunk 3 above)
Comment (tsunghanliu) [reply sanity/agent/deploy.py:493]: It's used to separate these two sentences.

File: sanity/agent/deploy.py (hunk 1 above)
Comment (tsunghanliu) [reply sanity/agent/deploy.py:192]: the new commit rewrite the usage of `send_mail()`

File: sanity/agent/agent.py (hunk 2 above)
Comment (kiya956) [reply sanity/agent/agent.py]: yes, thanks. Just update the PR

**Review by tsunghanliu — APPROVED** [review]
LGTM

**Pull request merged**